### PR TITLE
Added  the algorithm for Adjacency List

### LINKED ISSUE #1876: [Idea]: Adjacency List
### Idea Title

Adjacency List

### New Algorithm
 Write a program to store and display a graph in the form of adjacency list.
Let’s assume there are n vertices in the graph So, create an array of list of size n as adjList[n].
- adjList[0] will have all the nodes which are connected (neighbour) to vertex 0.
- adjList[1] will have all the nodes which are connected (neighbour) to vertex 1 and so on.

### Input:
The adjacent nodes of the graphs with which the curent node is connected.

### Output:
The corresponding Adjacency List.

### Potential Benefits

- [X] Improves user experience
- [X] Enhances learning
- [X] Adds a new feature
- [X] Improves code quality
- [X] Other

### Implementation Suggestions (Optional)

_No response_

### PR DESCRIPTION
**Pull request by shuvojitss**
### Description
Added the Adjacency List program under Graph.This program is w.r.t issue no. #1876 which is related to this program.

Fixes #1876

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/graphs/Adjacency-List.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+Enter the neighbour 1 of node 4: 3                                                              
+
+### Sample Output:
+
+The adjacency list is given by:                     
+0-->1-->2                  
+1-->0-->3                      
+2-->0-->3                      
+3-->2-->1-->4                   
+4-->3
```
Comment (ajay-dhangar) [inline docs/graphs/Adjacency-List.md]: write within ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/graphs/Adjacency-List.md (hunk 1 above)
Comment (shuvojitss) [reply docs/graphs/Adjacency-List.md]: Done

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**